--- a/sh-opc.h	2009-01-13 16:36:33.185640000 +0100
+++ b/sh-opc.h	2009-01-13 17:04:14.159687000 +0100
@@ -579,7 +579,7 @@
 
 /* 0110nnnnmmmm0000 mov.b @<REG_M>,<REG_N>*/{"mov.b",{A_IND_M,A_REG_N},{HEX_6,REG_N,REG_M,HEX_0}, arch_sh_up},
 
-/* 10000000mmmmi4*1 mov.b R0,@(<disp>,<REG_M>)*/{"mov.b",{A_R0,A_DISP_REG_M},{HEX_8,HEX_0,REG_M,IMM1_4}, arch_sh_up},
+/* 10000000mmmmi4*1 mov.b R0,@(<disp>,<REG_N>)*/{"mov.b",{A_R0,A_DISP_REG_N},{HEX_8,HEX_0,REG_N,IMM1_4}, arch_sh_up},
 
 /* 11000000i8*1.... mov.b R0,@(<disp>,GBR)*/{"mov.b",{A_R0,A_DISP_GBR},{HEX_C,HEX_0,IMM1_8}, arch_sh_up},
 
@@ -635,7 +635,7 @@
 
 /* 0110nnnnmmmm0001 mov.w @<REG_M>,<REG_N>*/{"mov.w",{A_IND_M,A_REG_N},{HEX_6,REG_N,REG_M,HEX_1}, arch_sh_up},
 
-/* 10000001mmmmi4*2 mov.w R0,@(<disp>,<REG_M>)*/{"mov.w",{A_R0,A_DISP_REG_M},{HEX_8,HEX_1,REG_M,IMM1_4BY2}, arch_sh_up},
+/* 10000001mmmmi4*2 mov.w R0,@(<disp>,<REG_N>)*/{"mov.w",{A_R0,A_DISP_REG_N},{HEX_8,HEX_1,REG_N,IMM1_4BY2}, arch_sh_up},
 
 /* 11000001i8*2.... mov.w R0,@(<disp>,GBR)*/{"mov.w",{A_R0,A_DISP_GBR},{HEX_C,HEX_1,IMM1_8BY2}, arch_sh_up},
 
